### PR TITLE
Uploading the code coverage report and Gas usage report

### PR DESCRIPTION
**Pull request by bidyut-arianelabs**
Depends on https://github.com/bosonprotocol/contracts/pull/274

## 1. Code coverage report uploaded.
---
Screenshot 2021-11-09_18-04-08-571

### Once downloaded, the Code coverage report file can be viewed in the browser like this.
Screenshot 2021-11-09_18-11-38-581

---
## 2. Gas report uploaded (Generated whenever the unit test is executed).
---
Screenshot 2021-11-09_18-03-43-936

### Once downloaded, the Gas reporter file can be viewed in the terminal using the `cat` command like this.
Screenshot 2021-11-09_18-05-54-048

---

### CODE REVIEW

**Review by cliffhall — CHANGES_REQUESTED** [review]
You see a dashes in the rightmost column of the gas report because it doesn't have a fiat ETH price to extend gas usage. 

I would like to see the unit test / gas reporting process use a coinmarketcap key so that the current price of ETH is shown and the cost of each method call and deployment is extended to a fiat price.

**Review by hswopeams — DISMISSED** [review]

**Comment by NFTanaconda** [comment]
nice

**Review by cliffhall — DISMISSED** [review]

**Comment by bidyut-arianelabs** [comment]
PR is now ready for review.
Data from coinmarketcap is now visible. refer https://github.com/bosonprotocol/contracts/actions/runs/1654441004

**Review by hswopeams — DISMISSED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -13,6 +13,7 @@ jobs:
     env:
       PROTOCOL_DEPLOYER_PRIVATE_KEY: 123456789abcdef123456789abcdef123456789abcdef123456789abcdef1234
       CC_TOKEN_DEPLOYER_PRIVATE_KEY: 123456789abcdef123456789abcdef123456789abcdef123456789abcdef1234
+      CMC_API_KEY: ${{ secrets.CMC_API_KEY_NEW }}
```
Comment (cliffhall) [inline .github/workflows/ci.yaml:16]: So we determined that there was some clobbering or mismatch of the key in the secrets object. But can we change the variable to be something like COINMARKETCAP_API_KEY? The names above it are verbose, so it doesn't need to be abbreviated or have NEW in it.

**Review by cliffhall — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (bidyut-arianelabs) [reply .github/workflows/ci.yaml]: I agree. I have renamed the secret variable to `COINMARKETCAP_API_KEY`. PR is now ready for review. @cliffhall

**Review by hswopeams — APPROVED** [review]

**Review by cliffhall — APPROVED** [review]
LGTM! 👍

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (cliffhall) [reply .github/workflows/ci.yaml]: Cool, thanks

**Pull request merged**